Classes/Indexer/NodeIndexer.php

```diff
@@ -430,6 +537,72 @@ public function updateIndexAlias(): void
         $this->indexDriver->aliasActions($aliasActions);
     }
 
+    /**
+     * Update the main alias to allow to query all indices at once
+     */
+    public function updateMainAlias()
+    {
+        $aliasActions = [];
+        $aliasNamePrefix = $this->searchClient->getIndexNamePrefix(); // The alias name is the unprefixed index name
+
+        $indexNames = $this->indexDriver->indexesByPrefix($aliasNamePrefix);
+        $indexNames = \array_values(\array_filter($indexNames, function ($indexName) {
+            $suffix = '-' . $this->indexNamePostfix;
+            $indexNameParts = \explode('-', $indexName);
+            return substr($indexName, 0 - strlen($suffix)) === $suffix && count($indexNameParts) === 3;
```

Comment:
If the configured index name itself contains a dash, counting the indexNameParts does not work as expected and the main alias can't be updated.

Reply:
@theilm, should be fixed

Reply:
@daniellienert I reviewed your commit and tested it, it works now! Thanks! :+1: